### ISSUE TITLE
solution set of sin(x)=0 is not simplified, presented as union of two sets

### ISSUE THREAD

**Issue by rzeta0** [sympy/sympy#25101]
The following sets up an equation $sin(x)=0$

`eq = sympy.Eq(sympy.sin(x), 0)`

The following asks simpy's solveset solver to solve this equation.

`sympy.solveset(eq, x)`

The result is:

`{2nπ∣n∈Z}∪{2nπ+π∣n∈Z}`

This could be simplified. My attempt is as follows:

`sympy.solveset(eq, x).simplify()`

which gives...

`{2nπ∣n∈Z}∪{π(2n+1)∣n∈Z}`

This is still not as simple as **{nπ∣n∈Z}**

Is it worth fixing this for a very common test example $sin(x)=0$ ?

**Comment by oscarbenjamin** [comment]
Yes, it is worth fixing. There are definitely existing PRs e.g. gh-21196, gh-18489, gh-17838, gh-17079, gh-11188.

We don't seem to have an existing issue for this though (at least not that is referenced from any of those PRs) so I think that we can keep this open.

The basic problem with each of those PRs is that they try to fix the problem in the wrong place. Firstly it would be useful to be able to simplify the output with `.simplify` but that simplification is not something that should happen automatically. Each of those PRs tried to handle this in the `Union` automatic evaluation code which is the wrong place.

Secondly the approach taken in those PRs is based on matching pairs of image sets in a union which fails if there are 3 imagesets to be combined e.g.:
```python
In [6]: S = Union(*(imageset(lambda n: 3*n+m, Integers) for m in range(3)))

In [7]: S
Out[7]: {3⋅n │ n ∊ ℤ} ∪ {3⋅n + 1 │ n ∊ ℤ} ∪ {3⋅n + 2 │ n ∊ ℤ}
```

Lastly while it would be nice if `.simplify` works in cases like this really the `solveset` code is at fault for generating this solution in the first place. Somewhere there should be some checking to see if a solution of this form is about to be generated and a better form should be given in the first place rather than trying to simplify after the fact.
